src/Microsoft.AspNet.Mvc.Core/ModelBinding/FormCollectionModelBinder.cs

```diff
@@ -30,19 +29,11 @@ public class FormCollectionModelBinder : IModelBinder
             if (request.HasFormContentType)
             {
                 var form = await request.ReadFormAsync();
-                if (bindingContext.ModelType.GetTypeInfo().IsAssignableFrom(form.GetType().GetTypeInfo()))
```

Comment:
:ok:

I wasn't suggesting any activation at all, just extending the compatibilites. But I understand we don't want to support anything derived from `IFormCollection`, even if the type is compatible with the instance `await request.ReadFormAsync()` returns. In any case my approach ends up looking more cumbersome:
``` c#
    object model;
    var request = bindingContext.OperationBindingContext.HttpContext.Request;
    if (request.HasFormContentType)
    {
        var form = await request.ReadFormAsync();
        if (!bindingContext.ModelType.IsAssignableFrom(form.GetType())
        {
            return null;
        }

        model = form;
    }
    else
    {
        if (!bindingContext.ModelType.IsAssignableFrom(typeof(EmptyFormCollection))
        {
            return null;
        }

        model = new EmptyFormCollection();
    }
```